[JSC][32bit] ASSERTION FAILED: !initialBytes

mikhailramalho opened the pull request:
#### 3866ba779bf1f21626f945d9003d005f1c78a3eb
<pre>
[JSC][32bit] ASSERTION FAILED: !initialBytes
<a href="https://bugs.webkit.org/show_bug.cgi?id=243901">https://bugs.webkit.org/show_bug.cgi?id=243901</a>

Reviewed by Yusuke Suzuki.

This patch fixes a value being truncated in 32bits platforms, resulting
in a invalid maximum size.

The method bytes() returns a uint64_t but in Memory::tryCreate, it is
stored in size_t, which is fine in 64bit platforms but gets truncated in
32 bits ones.

In this patch both initialBytes and maximumBytes are now uint64_t and
there is an extra check if the maximum size is greater than what&apos;s
allowed in the platform.

* Source/JavaScriptCore/wasm/WasmMemory.cpp:
(JSC::Wasm::Memory::tryCreate):

Canonical link: <a href="https://commits.webkit.org/253399@main">https://commits.webkit.org/253399@main</a>
</pre>

Constellation commented on Source/JavaScriptCore/wasm/WasmMemory.cpp:
```diff
@@ -386,12 +386,15 @@ RefPtr<Memory> Memory::tryCreate(VM& vm, PageCount initial, PageCount maximum, M
     if (initialBytes > MAX_ARRAY_BUFFER_SIZE)
         return nullptr; // Client will throw OOMError.
 
+    if (maximumBytes > MAX_ARRAY_BUFFER_SIZE)
```
Can you ensure the other 64bit engines' behavior on this? If maximum does not have a limit on 64bit environment, then we should not introduce this check for 64bit build.

mikhailramalho replied:
fixed

Constellation reviewed (approved):
r=me

mikhailramalho committed:
[JSC][32bit] ASSERTION FAILED: !initialBytes
https://bugs.webkit.org/show_bug.cgi?id=243901

Reviewed by Yusuke Suzuki.

This patch fixes a value being truncated in 32bits platforms, resulting
in a invalid maximum size.

The method bytes() returns a uint64_t but in Memory::tryCreate, it is
stored in size_t, which is fine in 64bit platforms but gets truncated in
32 bits ones.

In this patch both initialBytes and maximumBytes are now uint64_t and
there is an extra check if the maximum size is greater than what's
allowed in the platform.

* Source/JavaScriptCore/wasm/WasmMemory.cpp:
(JSC::Wasm::Memory::tryCreate):

Canonical link: https://commits.webkit.org/253399@main
